pyiron_contrib/workflow/channels.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any, Optional, TYPE_CHECKING
+from warnings import warn
+
+if TYPE_CHECKING:
+    from pyiron_contrib.workflow.node import Node
+
+
+class ChannelTemplate:
+    def __init__(
+            self,
+            name: str,
+            default: Optional[Any] = None,
+            types: Optional[tuple | type[Any]] = None,
+            storage_priority: int = 0,
+    ):
+        self.name = name
+        self.default = default
+        self.types = types
+        self.storage_priority = storage_priority
+
+    def _to_IOChannel(self, node: Node, class_: type[Channel]) -> Channel:
+        return class_(
+            name=self.name,
+            node=node,
+            default=deepcopy(self.default),
+            types=self.types,
+            storage_priority=self.storage_priority
+        )
+
+    def to_input(self, node: Node) -> InputChannel:
+        return self._to_IOChannel(node=node, class_=InputChannel)
+
+    def to_output(self, node: Node) -> OutputChannel:
+        return self._to_IOChannel(node=node, class_=OutputChannel)
+
+
+class Channel(ABC):
+    def __init__(
+            self,
+            name: str,
+            node: Node,
+            default: Optional[Any] = None,
+            types: Optional[tuple] = None,
+            storage_priority: int = 0,
+    ):
+        self.name = name
+        self.node = node
+        self.default = default
+        self.value = default
+        self.types = None if types is None else self._types_to_tuple(types)
+        self.storage_priority = storage_priority
+        self.connections = []
+
+    @staticmethod
```

Comment:
In my opinion this should not be needed, because if we use `typing` it takes care of whether it's one type or multiple types, i.e. the user should write either one type e.g. `int` or multiple types following `typing`, e.g. `int | float`